pkg/webhook/core.oam.dev/v1alpha2/application/validating_handler.go

```diff
@@ -85,9 +85,10 @@ func (h *ValidatingHandler) Handle(ctx context.Context, req admission.Request) a
 		if err := h.Decoder.DecodeRaw(req.AdmissionRequest.OldObject, oldApp); err != nil {
 			return admission.Errored(http.StatusBadRequest, err)
 		}
-
-		if allErrs := h.ValidateUpdate(ctx, app, oldApp); len(allErrs) > 0 {
-			return admission.Errored(http.StatusUnprocessableEntity, allErrs.ToAggregate())
+		if app.ObjectMeta.DeletionTimestamp.IsZero() {
+			if allErrs := h.ValidateUpdate(ctx, app, oldApp); len(allErrs) > 0 {
```

Comment:
```suggestion
			if len(h.ValidateUpdate(ctx, app, oldApp)) > 0 {
```
Better? :)

Reply:
This cannot be modified here, because the following logic needs to return the error in `allErrs`